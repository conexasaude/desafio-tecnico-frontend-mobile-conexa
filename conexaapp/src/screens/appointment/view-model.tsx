import {useEffect, useState} from 'react';
import {useAppContext} from '@hooks/use-app.hook';
import {IAppointment} from '@interfaces/appointment.interface';
import {useNavigation, useRoute} from '@react-navigation/native';
import {AppNavigatorRoutesProps} from '@routes/app.routes';
import {fetchAppointmentById} from '@infra/http/repositories/appointment.repository';

interface RouteParamsProps {
  id: number;
}

export interface AppointmentViewModel {
  appointment: IAppointment;
  handleGoBack: () => void;
}

function useAppointmentViewModel(): AppointmentViewModel {
  const {goBack} = useNavigation<AppNavigatorRoutesProps>();
  const route = useRoute();
  const {id} = route.params as RouteParamsProps;

  const {setIsLoading, onOpenDialog} = useAppContext();
  const [appointment, setAppointment] = useState<IAppointment>(
    {} as IAppointment,
  );

  async function getAppointmentById() {
    try {
      setIsLoading(true);
      const {data} = await fetchAppointmentById({id});
      setAppointment(data);
      setIsLoading(false);
    } catch (error) {
      onOpenDialog({message: error.message, title: 'Erro'});
      setIsLoading(false);
    }
  }

  function handleGoBack() {
    goBack();
  }

  useEffect(() => {
    getAppointmentById();
  }, []);

  return {
    appointment,
    handleGoBack,
  };
}

export {useAppointmentViewModel};
