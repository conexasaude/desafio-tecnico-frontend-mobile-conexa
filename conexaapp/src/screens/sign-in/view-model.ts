import {useState} from 'react';
import {yupResolver} from '@hookform/resolvers/yup';
import * as yup from 'yup';
import {
  Control,
  FieldErrors,
  FieldValues,
  useForm,
  UseFormHandleSubmit,
} from 'react-hook-form';
import {useAuthContext} from '@hooks/use-auth.hook';
import {useAppContext} from '@hooks/use-app.hook';

const signInSchema = yup.object({
  email: yup.string().required('Informe o email.').email('E-mail inv√°lido.'),
  password: yup.string().required('Informe a senha.'),
});

type FormDataProps = yup.InferType<typeof signInSchema> & FieldValues;

export interface SignInViewModel {
  control: Control<FormDataProps>;
  handleSubmit: UseFormHandleSubmit<FormDataProps>;
  errors: FieldErrors<FormDataProps>;
  passwordSecureTextEntry: boolean;
  handleSignIn: ({password, email}: FormDataProps) => Promise<void>;
  handlePasswordSecureTextEntry: () => void;
}

function useSignInViewModel(): SignInViewModel {
  const {signInUser} = useAuthContext();
  const {setIsLoading, onOpenDialog} = useAppContext();

  const {
    control,
    handleSubmit,
    formState: {errors},
  } = useForm<FormDataProps>({
    resolver: yupResolver(signInSchema),
  });

  const [passwordSecureTextEntry, setPasswordSecureTextEntry] = useState(true);

  async function handleSignIn({password, email}: FormDataProps) {
    try {
      setIsLoading(true);
      await signInUser(email, password);
    } catch (error) {
      onOpenDialog({message: error.message, title: 'Erro'});
      setIsLoading(false);
    }
  }

  function handlePasswordSecureTextEntry() {
    setPasswordSecureTextEntry(prev => !prev);
  }

  return {
    control,
    errors,
    passwordSecureTextEntry,
    handleSubmit,
    handleSignIn,
    handlePasswordSecureTextEntry,
  };
}

export {useSignInViewModel};
