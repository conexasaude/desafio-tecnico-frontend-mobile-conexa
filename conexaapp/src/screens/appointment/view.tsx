import {Container} from '@components/container/view';
import {Header} from '@components/header';
import BackIconSVG from '@assets/chevron-left.svg';
import {
  Card,
  Divider,
  Field,
  FieldBold,
  Icon,
  Title,
  styleSheet,
} from './styles';
import {useAppointmentViewModel} from './view-model';

export function Appointment() {
  const {appointment, handleGoBack} = useAppointmentViewModel();

  return (
    <Container>
      <Header.Root>
        <Header.VStack>
          <Header.Title>Detalhes</Header.Title>
          <Header.HStack style={styleSheet.HStack}>
            <Header.Button
              title="Voltar"
              leftElement={<BackIconSVG />}
              onPress={handleGoBack}
            />
            <Icon source={require('@assets/logo-icon-conexa.png')} />
          </Header.HStack>
        </Header.VStack>
      </Header.Root>

      {appointment?.id && (
        <Card>
          <Title>Dados do Paciente</Title>

          <Field>
            <FieldBold>Data da consulta: </FieldBold>
            {appointment.dataConsulta.split(' ')[0].replaceAll('-', '/')}
          </Field>
          <Field>
            <FieldBold>Hora da consulta: </FieldBold>
            {appointment.dataConsulta.split(' ')[1]}
          </Field>
          <Field>
            <FieldBold>Paciente: </FieldBold>
            {appointment.paciente}
          </Field>
          <Field>
            <FieldBold>Observações: </FieldBold>
            {appointment.observacao}
          </Field>

          <Divider />

          <Title>Dados do Profissional</Title>

          <Field>
            <FieldBold>Médico: </FieldBold>
            {appointment.medico.nome}
          </Field>
          {appointment.medico?.email && (
            <Field>
              <FieldBold>Contato: </FieldBold>
              {appointment.medico.email}
            </Field>
          )}
        </Card>
      )}
    </Container>
  );
}
