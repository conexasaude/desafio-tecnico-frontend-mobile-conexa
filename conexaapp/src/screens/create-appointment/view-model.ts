import {yupResolver} from '@hookform/resolvers/yup';
import * as yup from 'yup';
import {
  Control,
  FieldErrors,
  FieldValues,
  useForm,
  UseFormHandleSubmit,
} from 'react-hook-form';
import {useAuthContext} from '@hooks/use-auth.hook';
import {useAppContext} from '@hooks/use-app.hook';
import {UserModel} from '@models/user.model';
import {useNavigation} from '@react-navigation/native';
import {AppNavigatorRoutesProps} from '@routes/app.routes';
import dayjs from 'dayjs';
import {DateMap} from '@mappers/date.map';
import utc from 'dayjs/plugin/utc';
import timezone from 'dayjs/plugin/timezone';
import {createAppointment} from '@infra/http/repositories/appointment.repository';

dayjs.extend(utc);
dayjs.extend(timezone);

dayjs.tz.setDefault('America/Sao_Paulo');

const createAppointmentSchema = yup.object({
  patient: yup.string().required('Campo obrigatório.'),
  date: yup.string().required('Campo obrigatório.'),
  hour: yup.string().required('Campo obrigatório.'),
  description: yup.string().required('Campo obrigatório.'),
});

type FormDataProps = yup.InferType<typeof createAppointmentSchema> &
  FieldValues;

export interface CreateAppointmentViewModel {
  user: UserModel;
  control: Control<FormDataProps>;
  errors: FieldErrors<FormDataProps>;
  handleSubmit: UseFormHandleSubmit<FormDataProps>;
  handleCreate: ({
    patient,
    date,
    hour,
    description,
  }: FormDataProps) => Promise<void>;
  handleGoBack: () => void;
}

function useCreateAppointmentViewModel(): CreateAppointmentViewModel {
  const {goBack, reset} = useNavigation<AppNavigatorRoutesProps>();
  const {user} = useAuthContext();
  const {setIsLoading, onOpenDialog} = useAppContext();

  const {
    control,
    handleSubmit,
    formState: {errors},
  } = useForm<FormDataProps>({
    resolver: yupResolver(createAppointmentSchema),
  });

  async function handleCreate({
    patient,
    date,
    hour,
    description,
  }: FormDataProps) {
    const dateServer = DateMap.toServer(date);
    const convertedDate = dayjs(dateServer, 'YYYY/MM/DD');
    const currentDate = dayjs().format('YYYY/MM/DD');
    const isDateInPast = convertedDate.isBefore(currentDate);
    const isSameDate = convertedDate.isSame(currentDate);

    const now = dayjs();
    const hourDayjs = dayjs(
      `${convertedDate.format('YYYY/MM/DD')} ${hour}`,
      'YYYY/MM/DD HH:mm',
    );
    const isSameHour = hourDayjs.isSame(now);
    const isHourInPast = hourDayjs.isBefore(now);

    if (isDateInPast) {
      onOpenDialog({
        title: 'Erro',
        message: 'A data não pode ser inferior à data atual.',
      });
      return;
    }

    if (isSameDate && (isHourInPast || isSameHour)) {
      onOpenDialog({
        title: 'Erro',
        message: 'A hora não pode ser inferior à hora atual.',
      });
      return;
    }

    try {
      setIsLoading(true);
      await createAppointment({
        idMedico: 12,
        paciente: patient,
        dataConsulta: `${dateServer}T${hour}`,
        observacao: description,
      });

      setIsLoading(false);
      onOpenDialog({
        title: 'Sucesso',
        message: `Agendamento Efetuado com sucesso!\nO paciente ${patient} está agendado para ${date} às ${hour}.`,
      });
      reset({
        index: 0,
        routes: [{name: 'home'}],
      });
    } catch (error) {
      onOpenDialog({message: error.message, title: 'Erro'});
      setIsLoading(false);
    }
  }

  function handleGoBack() {
    goBack();
  }

  return {
    user,
    control,
    errors,
    handleSubmit,
    handleCreate,
    handleGoBack,
  };
}

export {useCreateAppointmentViewModel};
