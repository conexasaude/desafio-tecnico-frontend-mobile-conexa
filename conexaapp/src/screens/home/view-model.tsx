import {useEffect, useState} from 'react';
import {useAppContext} from '@hooks/use-app.hook';
import {IAppointment} from '@interfaces/appointment.interface';
import {fetchAppointments} from '@infra/http/repositories/appointment.repository';
import {useAuthContext} from '@hooks/use-auth.hook';
import {UserModel} from '@models/user.model';
import {useNavigation} from '@react-navigation/native';
import {AppNavigatorRoutesProps} from '@routes/app.routes';

export interface HomeViewModel {
  user: UserModel;
  appointments: IAppointment[];
  refresh: boolean;
  getAppointments: (refresh: boolean) => Promise<void>;
  handleSignOut: () => Promise<void>;
  handleCreateAppointment: () => void;
}

function useHomeViewModel(): HomeViewModel {
  const {navigate} = useNavigation<AppNavigatorRoutesProps>();
  const {user} = useAuthContext();
  const {setIsLoading, onOpenDialog} = useAppContext();

  const [refresh, setRefresh] = useState(false);
  const [appointments, setAppointments] = useState<IAppointment[]>(
    [] as IAppointment[],
  );

  async function getAppointments(refresh: boolean = false) {
    try {
      if (refresh) {
        setRefresh(true);
      } else {
        setIsLoading(true);
      }
      const {data} = await fetchAppointments();
      setAppointments(data);
      setIsLoading(false);
      setRefresh(false);
    } catch (error) {
      onOpenDialog({message: error.message, title: 'Erro'});
      setIsLoading(false);
    }
  }

  async function handleSignOut() {
    try {
      setIsLoading(true);
      await user.signOut();
      setIsLoading(false);
    } catch (error) {
      onOpenDialog({message: error.message, title: 'Erro'});
      setIsLoading(false);
    }
  }

  function handleCreateAppointment() {
    navigate('create-appointment');
  }

  useEffect(() => {
    getAppointments();
  }, []);

  return {
    user,
    appointments,
    refresh,
    getAppointments,
    handleSignOut,
    handleCreateAppointment,
  };
}

export {useHomeViewModel};
