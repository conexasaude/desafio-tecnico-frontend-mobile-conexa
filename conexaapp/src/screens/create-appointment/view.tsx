import {Platform, ScrollView, View} from 'react-native';
import {
  Form,
  FormContainer,
  Title,
  FormTitle,
  KeyboardAvoidingView,
  styleSheet,
  Icon,
  DateTimeContainer,
} from './styles';
import {Button} from '@components/button/view';
import {useCreateAppointmentViewModel} from './view-model';
import {InputForm} from '@components/form/input-form/view';
import {Container} from '@components/container/view';
import {Header} from '@components/header';
import BackIconSVG from '@assets/chevron-left.svg';
import {DateMask} from '@helpers/masks/date';
import {HourMask} from '@helpers/masks/hour';

export function CreateAppointment() {
  const {user, control, handleSubmit, errors, handleCreate, handleGoBack} =
    useCreateAppointmentViewModel();

  return (
    <KeyboardAvoidingView
      behavior={Platform.OS === 'ios' ? 'padding' : undefined}>
      <Header.Root>
        <Header.VStack>
          <Header.Title>Criar Agendamento</Header.Title>
          <Header.HStack style={styleSheet.HStack}>
            <Header.Button
              title="Voltar"
              leftElement={<BackIconSVG />}
              onPress={handleGoBack}
            />
            <Icon source={require('@assets/logo-icon-conexa.png')} />
          </Header.HStack>
        </Header.VStack>
      </Header.Root>

      <ScrollView
        showsVerticalScrollIndicator={false}
        keyboardShouldPersistTaps="always">
        <Container>
          <Title>Agendamento para {user.nome}</Title>
          <FormContainer>
            <Form>
              <FormTitle>Dados do Paciente</FormTitle>
              <InputForm
                control={control}
                name="patient"
                placeholder="jose@gmail.com"
                autoCapitalize="words"
                autoCorrect={false}
                error={errors.patient && String(errors.patient.message)}
                label="Nome do Paciente"
              />
              <DateTimeContainer>
                <View style={{flex: 1}}>
                  <InputForm
                    control={control}
                    name="date"
                    label="Data"
                    placeholder="Ex.: 10/12/2023"
                    autoCapitalize="none"
                    error={errors.date && String(errors.date.message)}
                    insertMask={DateMask}
                  />
                </View>
                <View style={{flex: 1}}>
                  <InputForm
                    control={control}
                    name="hour"
                    label="Hora"
                    placeholder="Ex.: 12:30"
                    autoCapitalize="none"
                    error={errors.hour && String(errors.hour.message)}
                    insertMask={HourMask}
                  />
                </View>
              </DateTimeContainer>

              <InputForm
                control={control}
                name="description"
                placeholder="Digite aqui as observações do paciente."
                autoCapitalize="sentences"
                autoCorrect={false}
                error={errors.description && String(errors.description.message)}
                label="Observações"
                multiline={true}
                style={{minHeight: 50}}
              />

              <Button
                title="Agendar"
                onPress={handleSubmit(handleCreate)}
                variant="outline"
              />
            </Form>
          </FormContainer>
        </Container>
      </ScrollView>
    </KeyboardAvoidingView>
  );
}
