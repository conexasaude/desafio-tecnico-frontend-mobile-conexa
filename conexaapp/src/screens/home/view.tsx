import {useHomeViewModel} from './view-model';
import {FlatList, RefreshControl} from 'react-native';
import {Appointment} from '@components/appointment/view';
import {Container} from '@components/container/view';
import {Header} from '@components/header';
import {CreateButton, LogOutIcon} from './styles';
import LogoSVG from '@assets/logo-conexa-white.svg';
import AddCircleSVG from '@assets/add-circle.svg';
import {ListEmpty} from '@components/list-empty/view';

export function Home() {
  const {
    user,
    appointments,
    refresh,
    getAppointments,
    handleSignOut,
    handleCreateAppointment,
  } = useHomeViewModel();

  return (
    <Container>
      <Header.Root>
        <Header.VStack>
          <LogoSVG />
          <Header.Greetings>{user.nome}</Header.Greetings>
        </Header.VStack>
        <Header.Button
          title="Sair"
          rightElement={<LogOutIcon />}
          onPress={handleSignOut}
        />
      </Header.Root>

      <FlatList
        data={appointments}
        keyExtractor={item => String(item.id)}
        renderItem={({item}) => <Appointment data={item} />}
        contentContainerStyle={[
          appointments?.length === 0 && {flex: 1},
          {padding: 24, paddingBottom: 50},
        ]}
        showsVerticalScrollIndicator={false}
        refreshControl={
          <RefreshControl
            tintColor={'white'}
            refreshing={refresh}
            onRefresh={() => {
              getAppointments(true);
            }}
          />
        }
        ListEmptyComponent={() => (
          <ListEmpty
            message={
              'Nenhum registro encontrado.\n Que tal iniciar agora agora?'
            }
          />
        )}
      />
      <CreateButton onPress={handleCreateAppointment}>
        <AddCircleSVG fill="blue" width={50} height={50} />
      </CreateButton>
    </Container>
  );
}
