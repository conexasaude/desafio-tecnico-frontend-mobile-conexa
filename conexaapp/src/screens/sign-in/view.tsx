import {Platform, ScrollView} from 'react-native';
import {
  Form,
  FormContainer,
  FormTitle,
  PasswordButton,
  KeyboardAvoidingView,
  Logo,
} from './styles';
import {Button} from '@components/button/view';
import {useSignInViewModel} from './view-model';
import EyeCloseSvg from '@assets/eye-close.svg';
import EyeOpenSvg from '@assets/eye-open.svg';
import {InputForm} from '@components/form/input-form/view';
import {Container} from '@components/container/view';
import {Header} from '@components/header';

export function SignIn() {
  const {
    control,
    handleSubmit,
    errors,
    handleSignIn,
    passwordSecureTextEntry,
    handlePasswordSecureTextEntry,
  } = useSignInViewModel();

  return (
    <KeyboardAvoidingView
      behavior={Platform.OS === 'ios' ? 'padding' : undefined}>
      <ScrollView
        showsVerticalScrollIndicator={false}
        keyboardShouldPersistTaps="always">
        <Container>
          <Header.Root>
            <Header.VStack>
              <Logo />
              <Header.Title>Conecte-se à sua saúde</Header.Title>
            </Header.VStack>
          </Header.Root>

          <FormContainer>
            <Form>
              <FormTitle>Acesse sua conta</FormTitle>
              <InputForm
                control={control}
                name="email"
                placeholder="jose@gmail.com"
                autoCapitalize="none"
                autoCorrect={false}
                error={errors.email && String(errors.email.message)}
                label="E-mail"
              />
              <InputForm
                control={control}
                name="password"
                label="Senha"
                placeholder="Digite aqui sua senha"
                autoCapitalize="none"
                error={errors.password && String(errors.password.message)}
                secureTextEntry={passwordSecureTextEntry}
                rightElement={
                  <PasswordButton onPress={handlePasswordSecureTextEntry}>
                    {passwordSecureTextEntry ? (
                      <EyeCloseSvg width={20} height={20} />
                    ) : (
                      <EyeOpenSvg width={20} height={20} />
                    )}
                  </PasswordButton>
                }
              />

              <Button
                title="Acessar conta"
                onPress={handleSubmit(handleSignIn)}
                variant="outline"
              />
            </Form>
          </FormContainer>
        </Container>
      </ScrollView>
    </KeyboardAvoidingView>
  );
}
