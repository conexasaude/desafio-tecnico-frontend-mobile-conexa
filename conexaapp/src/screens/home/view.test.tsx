import {act, fireEvent, render} from '@testing-library/react-native';
import {Home} from './view';
import {HomeViewModel, useHomeViewModel} from './view-model';
import {ViewProviderWrapper} from '../../../jest-utils/wrapper';
import {UserModel} from '@models/user.model';
import {IAppointment} from '@interfaces/appointment.interface';
import AddCircleSVG from '@assets/add-circle.svg';

jest.mock('./view-model');
jest.mock('@assets/ios-log-out.svg', () => 'LogOutSVG');
jest.mock('@assets/add-circle.svg', () => 'AddCircleSVG');
jest.mock('@assets/logo-conexa-white.svg', () => 'LogoSvg');

interface CreateMockHomeView {
  user: UserModel;
  appointments: IAppointment[];
  refresh: boolean;
}

const createMockHomeViewModel = ({
  user,
  appointments,
  refresh,
}: CreateMockHomeView): HomeViewModel => ({
  user,
  appointments,
  refresh,
  getAppointments: jest.fn(),
  handleSignOut: jest.fn(),
  handleCreateAppointment: jest.fn(),
});

interface FactoryProps {
  user?: UserModel;
  appointments?: IAppointment[];
  refresh?: boolean;
}

const defaultMockedUser = {
  nome: 'John Doe',
  email: 'john.doe@gmail.com',
  signOut: jest.fn(),
} as unknown;

function makeSut({
  user = defaultMockedUser as UserModel,
  appointments = [] as IAppointment[],
  refresh = false,
}: FactoryProps) {
  const mockHomeViewModel = createMockHomeViewModel({
    user,
    appointments,
    refresh,
  });

  (useHomeViewModel as jest.Mock).mockReturnValueOnce(mockHomeViewModel);

  const sut = render(
    <ViewProviderWrapper>
      <Home />
    </ViewProviderWrapper>,
  );

  return {...sut, ...mockHomeViewModel};
}

describe('Home view', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });
  it('should render correctly', () => {
    const {toJSON} = makeSut({});

    expect(toJSON()).toMatchSnapshot();
  });

  it('should render correctly with user name', async () => {
    const {toJSON, findByText} = makeSut({});
    const mockedUser = defaultMockedUser as UserModel;
    const grettings = `OlÃ¡ ${mockedUser.nome}`;

    const grettingsText = await findByText(grettings);

    expect(grettingsText.props.children[1]).toEqual(mockedUser.nome);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render appointments correctly', () => {
    const mockedAppointments: IAppointment[] = [
      {
        id: 16,
        medico: {
          id: 1,
          nome: 'Dr. Daniel Vieira',
          email: null,
        },
        paciente: 'Diego Senna',
        dataConsulta: '2020-04-23 08:30',
        observacao: 'Exemplo de consulta',
      },
      {
        id: 11,
        medico: {
          id: 1,
          nome: 'Dr. Daniel Vieira',
          email: null,
        },
        paciente: 'Rafael Braga',
        dataConsulta: '2020-04-23 08:30',
        observacao: 'Exemplo de consulta',
      },
    ];

    const {queryByText} = makeSut({
      appointments: mockedAppointments,
    });

    mockedAppointments.forEach(appointment => {
      expect(queryByText(`Paciente: ${appointment.paciente}`)).toBeTruthy();
    });
  });

  it('should call handleSignOut when "Sair" button is pressed', async () => {
    const {findByText, handleSignOut} = makeSut({});

    const signOutButton = await findByText('Sair');

    await act(() => {
      fireEvent.press(signOutButton);
    });

    expect(handleSignOut).toHaveBeenCalledTimes(1);
  });

  it('should call handleCreateAppointment when Plus button is pressed', async () => {
    const {UNSAFE_getByType, handleCreateAppointment} = makeSut({});

    await act(() => {
      fireEvent.press(UNSAFE_getByType(AddCircleSVG));
    });

    expect(handleCreateAppointment).toHaveBeenCalled();
  });
});
