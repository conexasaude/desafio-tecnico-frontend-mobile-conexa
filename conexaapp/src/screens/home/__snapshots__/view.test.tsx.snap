// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home view should render correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#1800AE",
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
    }
  }
>
  <RCTSafeAreaView
    style={
      {
        "backgroundColor": "#1800AE",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
          "gap": 5,
          "justifyContent": "space-between",
          "paddingBottom": 10,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 10,
          "width": "100%",
        }
      }
    >
      <View
        style={
          {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "gap": 5,
          }
        }
      >
        <LogoSvg />
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 27,
            }
          }
        >
          <Text
            style={
              {
                "fontWeight": "bold",
              }
            }
          >
            Olá 
          </Text>
          John Doe
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 5,
            "opacity": 1,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 27,
            }
          }
        >
          Sair
        </Text>
        <LogOutSVG
          fill="#FFFFFF"
          height={25}
          style={{}}
          width={25}
        />
      </View>
    </View>
    <RCTScrollView
      ListEmptyComponent={[Function]}
      contentContainerStyle={
        [
          {
            "flex": 1,
          },
          {
            "padding": 24,
            "paddingBottom": 50,
          },
        ]
      }
      data={[]}
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      refreshControl={
        <RefreshControlMock
          onRefresh={[Function]}
          refreshing={false}
          tintColor="white"
        />
      }
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showsVerticalScrollIndicator={false}
      stickyHeaderIndices={[]}
      viewabilityConfigCallbackPairs={[]}
    >
      <RCTRefreshControl />
      <View>
        <View
          style={
            {
              "alignItems": "center",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              {
                "color": "#DDDEDF",
                "fontSize": 31,
                "textAlign": "center",
              }
            }
          >
            Nenhum registro encontrado.
 Que tal iniciar agora agora?
          </Text>
        </View>
      </View>
    </RCTScrollView>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#FFFFFF",
          "borderBottomLeftRadius": 28,
          "borderBottomRightRadius": 28,
          "borderTopLeftRadius": 28,
          "borderTopRightRadius": 28,
          "bottom": 0,
          "height": 54,
          "justifyContent": "center",
          "opacity": 1,
          "position": "absolute",
          "width": 54,
        }
      }
    >
      <AddCircleSVG
        fill="blue"
        height={50}
        width={50}
      />
    </View>
  </RCTSafeAreaView>
</View>
`;

exports[`Home view should render correctly with user name 1`] = `
<View
  style={
    {
      "backgroundColor": "#1800AE",
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
    }
  }
>
  <RCTSafeAreaView
    style={
      {
        "backgroundColor": "#1800AE",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
          "gap": 5,
          "justifyContent": "space-between",
          "paddingBottom": 10,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 10,
          "width": "100%",
        }
      }
    >
      <View
        style={
          {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "gap": 5,
          }
        }
      >
        <LogoSvg />
        <Text
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 27,
            }
          }
        >
          <Text
            style={
              {
                "fontWeight": "bold",
              }
            }
          >
            Olá 
          </Text>
          John Doe
        </Text>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "gap": 5,
            "opacity": 1,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            {
              "color": "#FFFFFF",
              "fontSize": 27,
            }
          }
        >
          Sair
        </Text>
        <LogOutSVG
          fill="#FFFFFF"
          height={25}
          style={{}}
          width={25}
        />
      </View>
    </View>
    <RCTScrollView
      ListEmptyComponent={[Function]}
      contentContainerStyle={
        [
          {
            "flex": 1,
          },
          {
            "padding": 24,
            "paddingBottom": 50,
          },
        ]
      }
      data={[]}
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      refreshControl={
        <RefreshControlMock
          onRefresh={[Function]}
          refreshing={false}
          tintColor="white"
        />
      }
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      showsVerticalScrollIndicator={false}
      stickyHeaderIndices={[]}
      viewabilityConfigCallbackPairs={[]}
    >
      <RCTRefreshControl />
      <View>
        <View
          style={
            {
              "alignItems": "center",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "justifyContent": "center",
            }
          }
        >
          <Text
            style={
              {
                "color": "#DDDEDF",
                "fontSize": 31,
                "textAlign": "center",
              }
            }
          >
            Nenhum registro encontrado.
 Que tal iniciar agora agora?
          </Text>
        </View>
      </View>
    </RCTScrollView>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "#FFFFFF",
          "borderBottomLeftRadius": 28,
          "borderBottomRightRadius": 28,
          "borderTopLeftRadius": 28,
          "borderTopRightRadius": 28,
          "bottom": 0,
          "height": 54,
          "justifyContent": "center",
          "opacity": 1,
          "position": "absolute",
          "width": 54,
        }
      }
    >
      <AddCircleSVG
        fill="blue"
        height={50}
        width={50}
      />
    </View>
  </RCTSafeAreaView>
</View>
`;
